modules/network/kademlia/kademlia.js

```diff
@@ -216,61 +190,64 @@ class Kademlia {
-                    'kad-encrypted-key-process-result',
-                    'kad-replication-request', 'kad-replacement-replication-request', 'kad-replacement-replication-finished',
-                ],
-                difficulty: this.config.network.solutionDifficulty,
-            }));
-            this.log.info('Hashcash initialised');
+            this.node.content = this.node.plugin(kadence.contentaddress({ valueEncoding: 'hex' }));
+            this.log.info('Content initialised');
 
-            this.node.blacklist = this.node.plugin(kadence.churnfilter({
-                cooldownBaseTimeout: this.config.network.churnPlugin.cooldownBaseTimeout,
-                cooldownMultiplier: this.config.network.churnPlugin.cooldownMultiplier,
-                cooldownResetTime: this.config.network.churnPlugin.cooldownResetTime,
-            }));
-            this.log.info('Churn filter initialised');
+            this.node.eclipse = this.node.plugin(kadence.eclipse(this.identity));
+            this.log.info('Eclipse initialised');
+
+            const peerCacheFilePath = path.join(
+                this.config.appDataPath,
+                this.config.embedded_peercache_path,
+            );
+
+            if (!fs.existsSync(peerCacheFilePath)) {
+                fs.writeFileSync(peerCacheFilePath, '{}');
+            }
+
+            this.node.rolodex = this.node.plugin(kadence.rolodex(peerCacheFilePath));
+            this.log.info('Rolodex initialised');
 
-            // Patch Churn to ignore all outgoing requests towards blacklisted contacts.
-            const send = this.node.send.bind(this.node);
-            this.node.send = function (method, params, target, handler) {
+            // Override node's _updateContact method to filter contacts.
+            this.node._updateContact = (identity, contact) => {
                 try {
-                    const contactId = target[0].toString('hex');
-                    if (this.node.blacklist.hasBlock(contactId)) {
-                        this.log.debug('Trying to send to blacklisted contact: %s.', contactId);
-                        return handler(Error('Contact blacklisted.'));
+                    if (!this.validateContact(identity, contact)) {
+                        this.log.debug(`Ignored contact ${identity}. Hostname ${contact.hostname}. Network ID ${contact.network_id}.`);
+                        return;
                     }
-                    send(method, params, target, handler);
-                } catch (e) {
-                    this.log.error('Failed to check for blacklist');
-                    handler(Error('Failed to check for blacklist.'));
+                } catch (err) {
+                    this.log.debug(`Failed to filter contact(${identity}, ${contact}). ${err}.`);
+                    return;
                 }
-            }.bind(this);
 
-            if (this.config.onion_enabled) {
-                this.enableOnion();
-            }
```

Comment:
Are we sure this is something we want to delete?

Reply:
this.enableOnion() does not exist in the current implementation.
We could add the following code from kadence.js:

```
// Use Tor for an anonymous overlay
  if (!!parseInt(config.OnionEnabled)) {
    kadence.constants.T_RESPONSETIMEOUT = 20000;
    node.onion = node.plugin(kadence.onion({
      dataDirectory: config.OnionHiddenServiceDirectory,
      virtualPort: config.OnionVirtualPort,
      localMapping: `127.0.0.1:${config.NodeListenPort}`,
      torrcEntries: {
        CircuitBuildTimeout: 10,
        KeepalivePeriod: 60,
        NewCircuitPeriod: 60,
        NumEntryGuards: 8,
        Log: `${config.OnionLoggingVerbosity} stdout`
      },
      passthroughLoggingEnabled: !!parseInt(config.OnionLoggingEnabled)
    }));
  }
```